src/Symfony/Component/HttpKernel/DependencyInjection/AddCacheClearerPass.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Registers the cache clearers.
+ *
+ * @author Dustin Dobervich <ddobervich@gmail.com>
+ */
+class AddCacheClearerPass implements CompilerPassInterface
+{
+    private $tag;
+    private $definition;
+
+    public function __construct($definition = null, $tag = null)
+    {
+        if (null === $definition || null === $tag) {
+            @trigger_error('The '.static::class.' class\' constructor needs two parameters since version 3.2, and will become required in 4.0.', E_USER_DEPRECATED);
+
+            $definition = 'cache_clearer';
```

Comment:
Yes, as I just learned that indeed `func_num_args` returns the number of args given to a function